zerver/tests/test_subs.py

```diff
@@ -1842,7 +1842,7 @@ def test_private_stream_subscription(self) -> None:
 
         self.assert_length(events, 3)
         create_event, add_event, add_peer_event = events
-
+        #
```

Comment:
To test the subscription message notification bot should be subscribed to stream. Should I subscribe bot to stream here and test or create a new test for this thing?

Reply:
I think we should have the notification bot use an API to send the message that doesn't require it be subscribed.

Reply:
We should use API to send the messages from notification bot in testing only. Right?